packages/flutter_tools/lib/src/macos/build_macos.dart

```diff
@@ -271,3 +283,52 @@ Future<void> _writeCodeSizeAnalysis(BuildInfo buildInfo, SizeAnalyzer? sizeAnaly
     'dart devtools --appSizeBase=$relativeAppSizePath'
   );
 }
+
+/// Finds and copies macOS entitlements file. In the copy, disables sandboxing.
+/// If entitlements file is not found, returns null.
+///
+/// As of macOS 14, running a macOS sandbox app may prompt the user to grant
+/// access to the app. To workaround this in CI, we create and use a entitlements
+/// file with sandboxing disabled. See
+/// https://developer.apple.com/documentation/security/app_sandbox/accessing_files_from_the_macos_app_sandbox.
+File? _createDisabledSandboxEntitlementFile(
```

Comment:
Instead of actually doing this in the tool, can we set `FLUTTER_XCODE_CODE_SIGN_ENTITLEMENTS` to the altered file in the test runner?

Reply:
I'm not quite sure what you mean. Do you mean create the altered file in the test_runner (whether that's devicelab or integration tests, etc) and then pass it in to the tool?

I think that'd be pretty difficult, because we call into the tool is many different ways so I wouldn't be confident that I got them all. For example, here's just a few example I found:
https://github.com/flutter/flutter/blob/4cad9fde67916cf43ef43057bb0224affd4e124a/packages/flutter_tools/test/integration.shard/transition_test_utils.dart#L161
https://github.com/flutter/flutter/blob/4cad9fde67916cf43ef43057bb0224affd4e124a/dev/devicelab/lib/framework/utils.dart#L501
https://github.com/flutter/flutter/blob/4cad9fde67916cf43ef43057bb0224affd4e124a/packages/flutter_tools/test/integration.shard/build_macos_config_only_test.dart#L27-L36

We also call into the tool from other repos (like [packages](https://github.com/flutter/packages/blob/24b0eb54a72dc94a33317f18aed367ffe3118c41/packages/pigeon/tool/shared/test_suites.dart#L246-L254))

Reply:
> I think that'd be pretty difficult, because we call into the tool is many different ways so I wouldn't be confident that I got them all. For example, here's just a few example I found:

Hmm, I guess I was thinking about `devicelab`.

Logic like this:
https://github.com/flutter/flutter/blob/4cad9fde67916cf43ef43057bb0224affd4e124a/dev/devicelab/lib/framework/utils.dart#L475-L479
But for the environment like here:
https://github.com/flutter/flutter/blob/4cad9fde67916cf43ef43057bb0224affd4e124a/dev/devicelab/lib/framework/utils.dart#L297


But I wasn't thinking of the `integration.shard`...


Bleeeeh I hate putting CI stuff in the tool.

Reply:
I agree it's not ideal, but I think the alternative would be to add logic everywhere we use `flutter run`, `flutter test`, `flutter drive` on macOS apps across our entire CI and repos. Also perhaps if customer testing had macOS integration tests, that could pose a problem (not that we currently have any that do, just another use case to consider).

Reply:
Since you're already working on this, we should consider the non-Google CI macOS test case.  I suspect most of the reports in https://github.com/flutter/flutter/issues/118469 are actually this issue (the OpenGL thing came at the very end).

Could we re-use your `--ci` flag to trigger this behavior, and print a warning like "Detected macOS app running in CI, turning off sandboxing."?

Reply:
However, if that's infeasible, this approach LGTM

Reply:
Sure, I updated to use either `--ci` or `LUCI_CI`. Also, updated message to tell users they can use `--ci` to get Flutter to attempt to disable sandboxing